### DISCUSSION TITLE
Remote Access

### DISCUSSION THREAD

**Discussion by crazynightgriffin** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#454] [Q&A]
I assume the only way to gain access when away from home would be to punch a hole in the router through to the ESP?

**Comment by Jay-Ti** [comment]
Yes and no. I'm facing the same Problem. 
It really depends on your network situation at the tub.
If you have a public IP, then yes, you could punch a hole.
You could also use a VPN and connect to the tubs local network(preferable to all other options),
Or you use MQTT on a public IP and connect both yourself and the tub to it. I've changed a bit of the MQTT code, so I can use a DDNS Service, since the code only allows for a static IP.

**Reply by crazynightgriffin** [reply]
Yeah sleeping on it I realised the mod had HA integration so could use that, cant wait to get started fingers crossed I can get all the parts quickly and dont blow my egg up.

**Reply by Jay-Ti** [reply]
If i didn't blow it up, you won't be able to. I've basically tried every pin config, and didn't fry it. Still wouldn't recommend doing that.

**Reply by uximxim** [reply]
> I've changed a bit of the MQTT code, so I can use a DDNS Service, since the code only allows for a static IP.

Jay-Ti, could you let me know what did you change in the code exactly? I was just looking how to achieve that as I have my MQTT broker on dynamic IP so DDNS is the only option to keep the connection after IP changes.

Thanks in advance.
